### ISSUE TITLE
Sum DPD Wigner-D functions over all allowed helicities

### ISSUE THREAD

**Issue by redeboer** [ComPWA/ampform#309]
## Problem

`DalitzPlotDecomposition` currently determines the summation domains of its alignment Wigner-$D$ functions from the spin projections present in the QRules transitions. Those projections can be restricted by the requested polarization and therefore do not necessarily form a complete spin basis.

For example, a massive spin-1 initial state generated with projections $\{-1,+1\}$ still requires the internal DPD helicity sum to run over $\{-1,0,+1\}$. Omitting $\lambda_0'=0$ makes the basis rotation incomplete. Conversely, the zero-helicity state must remain excluded for massless particles.

The allowed internal helicities should therefore be derived from each outer particle's spin and mass instead of collected from the generated transitions.

## Blocker

Expanding the summation domains exposes amplitude components such as $A^{ij}_{0,\ldots}$ that the current `HelicityAmplitudeBuilder` does not formulate when the corresponding projection was absent from the QRules reaction. This leaves unresolved indexed amplitudes in `HelicityModel.expression`. The amplitude-builder redesign tracked in #318 must provide the complete amplitude definitions required by these sums, so this issue is blocked by #318.

## Requirements

- Derive every DPD alignment summation domain from the corresponding particle's complete allowed helicity range.
- Exclude zero helicity for massless particles, using the same tolerant mass comparison as the other spin-alignment implementation.
- Ensure every indexed amplitude introduced by the expanded sums has a definition, so `HelicityModel.expression` contains no unresolved amplitude components.
- Cover a polarized massive spin-1 state whose QRules transitions contain only $\{-1,+1\}$ and verify that the internal sum also contains $0$.
- Remove the documentation warning that says the DPD sums omit $\lambda_0=0$ once the complete model can be formulated.

**Comment by redeboer** [comment]
Closing as not planned because the legacy in-package DPD implementation is being superseded by #504, which tracks integrating the complete AmpForm-DPD builder into AmpForm. The full internal helicity-basis requirement remains relevant to that integration; the spin-density-matrix work in #7 concerns external polarization and is separate.
